src/zarr/api/asynchronous.py

```diff
@@ -9,7 +9,7 @@
 import numpy.typing as npt
 from typing_extensions import deprecated
 
-from zarr.core.array import Array, AsyncArray, get_array_metadata
+from zarr.core.array import Array, AsyncArray, create_array, get_array_metadata
```

Comment:
Question: why did we put `create_array` in `zarr.core.array` and not in this module?

Reply:
if `create_array` is defined in `api.asynchronous`, then the `Group.create_array` method has to import `create_array` from `api.asynchronous`, but that module imports the `Group` class, which leads to a circular import. So I moved `create_array` lower in the hierarchy, which makes it easy for `group.py` to import.

Reply:
got it.